src/query/stream/stream_consumer_factory.cpp

```diff
@@ -0,0 +1,237 @@
+// Copyright 2024 Memgraph Ltd.
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.txt; by using this file, you agree to be bound by the terms of the Business Source
+// License, and you may not use this file except in compliance with the Business Source License.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0, included in the file
+// licenses/APL.txt.
+
+#include "query/stream/stream_consumer_factory.hpp"
+
+#include <memory>
+#include "query/discard_value_stream.hpp"
+#include "query/interpreter.hpp"
+#include "query/procedure/module.hpp"
+
+namespace memgraph::metrics {
+extern const Event MessagesConsumed;
+}  // namespace memgraph::metrics
+
+using memgraph::storage::Storage;
+using memgraph::storage::View;
+using memgraph::utils::MemoryResource;
+using memgraph::utils::NewDeleteResource;
+
+namespace memgraph::query::stream {
+
+namespace {
+
+inline constexpr auto kExpectedTransformationResultSize = 2;
+inline constexpr auto kCheckStreamResultSize = 2;
+
+constexpr std::string_view query_param_name{"query"};
+constexpr std::string_view params_param_name{"parameters"};
+
+std::pair<TypedValue /*query*/, TypedValue /*parameters*/> ExtractTransformationResult(
+    const utils::pmr::map<utils::pmr::string, TypedValue> &values, std::string_view transformation_name,
+    std::string_view stream_name) {
+  if (values.size() != kExpectedTransformationResultSize) {
+    throw StreamsException(
+        "Transformation '{}' in stream '{}' did not yield all fields (query, parameters) as required.",
+        transformation_name, stream_name);
+  }
+
+  auto get_value = [&](std::string_view field_name) mutable -> const TypedValue & {
```

Comment:
You can capture just values instead of everything

Reply:
Refactored by moving from elsewhere. So is at it was. But also it makes no difference. Values, transformation_name, and stream_name are all used inside the lambda. Also this is local use only, lambda would be completely inlined.